viewer/db.js

```diff
@@ -1730,61 +1750,118 @@ Db.getIndices = async (startTime, stopTime, bounding, rotateIndex) => {
+            index = queryExtraIndexTimeMatch[0];
+          }
+        }
 
-      switch (bounding) {
-      default:
-      case 'last':
-        if (stop >= startTime && start <= stopTime) {
-          indices.push(iname);
+        if (!queryExtraIndexTimeMatched){
+            // weekly 24w10                       v year     w  v week
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])[Ww]([0-4][0-9]|5[0-3])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
         }
-        break;
-      case 'first':
-      case 'both':
-      case 'either':
-      case 'database':
-        if (stop >= (startTime - len) && start <= (stopTime + len)) {
-          indices.push(iname);
+
+        if (!queryExtraIndexTimeMatched){
+            // monthly 24m10                      v year     w  v month
+          queryExtraIndexTimeMatch = iname.match(/([0-9][0-9])[Mm](0[1-9]|1[0-2])$/);
+          if (queryExtraIndexTimeMatch) {
+            queryExtraIndexTimeMatched = true;
+            index = queryExtraIndexTimeMatch[0];
+          }
         }
-        break;
+
+        // TODO: hourly2, ..., hourly12
+      } // if (isQueryExtraIndex)
+
+      if (!isQueryExtraIndex || queryExtraIndexTimeMatched) {
+
+        if (+index[0] >= 6) {
+          year = 1900 + (+index[0]) * 10 + (+index[1]);
+        } else {
+          year = 2000 + (+index[0]) * 10 + (+index[1]);
+        }
+
+        if (index[2] === 'w') {
+          len = 7 * 24 * 60 * 60;
+          month = 1;
+          day = (+index[3] * 10 + (+index[4])) * 7;
+        } else if (index[2] === 'm') {
+          month = (+index[3]) * 10 + (+index[4]);
+          day = 1;
+          len = 31 * 24 * 60 * 60;
+        } else if (index.length === 6) {
+          month = (+index[2]) * 10 + (+index[3]);
+          day = (+index[4]) * 10 + (+index[5]);
+          len = 24 * 60 * 60;
+        } else {
+          month = (+index[2]) * 10 + (+index[3]);
+          day = (+index[4]) * 10 + (+index[5]);
+          hour = (+index[7]) * 10 + (+index[8]);
+          len = hlength;
+        }
+
+        const start = Date.UTC(year, month - 1, day, hour) / 1000;
+        const stop = Date.UTC(year, month - 1, day, hour) / 1000 + len;
+
+        switch (bounding) {
+        default:
+        case 'last':
+          if (stop >= startTime && start <= stopTime) {
+            indices.push(iname);
+          }
+          break;
+        case 'first':
+        case 'both':
+        case 'either':
+        case 'database':
+          if (stop >= (startTime - len) && start <= (stopTime + len)) {
+            indices.push(iname);
+          }
+          break;
+        }
+
+      } else if (isQueryExtraIndex) {
+        // this is a extra user-specified index pattetern from queryExtraIndices, and
+        //   we couldn't grok it, so just query the whole thing
+        indices.push(iname);
```

Comment:
If we had a user-defined query index that didn't match the right suffix pattern(s), then I guess we need to query it anyway, although this will probably be worse performance (?). Don't do this, users.